packages/ckeditor5-table/src/utils/structure.js

```diff
@@ -305,3 +305,36 @@ function addHeadingsToCroppedTable( croppedTable, sourceTable, startRow, startCo
 		updateNumericAttribute( 'headingColumns', headingColumnsInCrop, croppedTable, writer, 0 );
 	}
 }
+
+/**
+ * Returns array of column indexes that have no cells anchored.
+ *
+ * For this table:
+ *
+ *     +----+----+----+----+----+----+----+
+ *     | 00 | 01      | 03 | 04      | 06 |
+ *     +----+----+----+----+         +----+
+ *     | 10 | 11      | 13 |         | 16 |
+ *     +----+----+----+----+----+----+----+
+ *     | 20 | 21      | 23 | 24      | 26 |
+ *     +----+----+----+----+----+----+----+
+ *                  ^--- empty ---^
+ *
+ * Will return: [ 2, 5 ]
+ */
+export function getEmptyColumnsIndexes( table ) {
+	const tableMap = Array.from( new TableWalker( table, { includeAllSlots: true } ) );
+	const lastColumn = tableMap[ tableMap.length - 1 ].column;
```

Comment:
I'm concerned about this - according to HTML table model docs the first row defines the length of a table. Here however it is probably needed to check all columns width. So maybe a comment on that is needed.

But if not, a `TableUtils.gerColumns()` + `TableWalker` without `includeAllSlots` should do the trick to get `lastColumn`